pkg/image-graph-generator/operator.go

```diff
@@ -80,12 +80,14 @@ func (o *Operator) callback(c *api.ReleaseBuildConfiguration, i *config.Info) er
 	}
 
 	var errs []error
-	excludedImages := sets.New[string](c.PromotionConfiguration.ExcludedImages...)
-
-	for _, image := range c.Images {
-		if !excludedImages.Has(string(image.To)) {
-			if err := o.UpdateImage(image, c, branchID); err != nil {
-				errs = append(errs, err)
+	for _, target := range api.PromotionTargets(c.PromotionConfiguration) {
+		excludedImages := sets.New[string](target.ExcludedImages...)
+
+		for _, image := range c.Images {
+			if !excludedImages.Has(string(image.To)) {
```

Comment:
This has to take into account the new meaning of `*`.

Reply:
The existing code also doesn't seem to handle `additional_images`.

Reply:
I have yet to code in the `*` bit - and as for `additional_images` here, I'm not sure exactly how to proceed, all I did in this diff was move the code into a loop over targets; so the previous code also did not handle `additional_images` - should it have?

Reply:
I was hoping we could just use the same code that's used in other places to determine the images to be promoted and get rid of this partial (re-)implementation.

Reply:
(Not impossible, just asking for this to be a follow-up that someone on DPTP could handle?)

Reply:
Leave a comment here too.  There's no reason for this code duplication.